frontend/projects/ui/src/app/services/connection.service.ts

```diff
@@ -5,66 +5,61 @@ import {
   fromEvent,
   merge,
   Observable,
+  Subject,
 } from 'rxjs'
-import { PatchConnection, PatchDbService } from './patch-db/patch-db.service'
-import {
-  distinctUntilChanged,
-  map,
-  mapTo,
-  startWith,
-  tap,
-} from 'rxjs/operators'
+import { distinctUntilChanged, map, tap } from 'rxjs/operators'
 import { ConfigService } from './config.service'
 
+export enum PatchConnection {
+  Initializing = 'initializing',
+  Connected = 'connected',
+  Disconnected = 'disconnected',
+}
+
 @Injectable({
   providedIn: 'root',
 })
 export class ConnectionService {
+  private readonly patchDBConnectionError$ = new Subject<any>()
+  private readonly patchConnection$ = new BehaviorSubject<PatchConnection>(
+    PatchConnection.Initializing,
+  )
   private readonly networkState$ = merge(
-    fromEvent(window, 'online').pipe(mapTo(true)),
-    fromEvent(window, 'offline').pipe(mapTo(false)),
+    fromEvent(window, 'online'),
+    fromEvent(window, 'offline'),
   ).pipe(
-    startWith(null),
```

Comment:
`startWith(null)` was here so that upon initial subscription it would run through following `map` and emit current network state.